.github/workflows/cmake.yml

```diff
@@ -42,20 +56,62 @@ jobs:
     - uses: actions/checkout@v3
       with:
         submodules: 'recursive'
+
     - uses: actions/setup-python@v4
       with:
         python-version: '3.10'
+
     - uses: BSFishy/pip-action@v1
       with:
         requirements: requirements.txt
 
+    - name: check_env
+      run: |
+        which python
+        python -m pip list
+
     - name: Build
-      shell: bash
       # Execute the build.  You can specify a specific target with "--target <NAME>"
-      run: python3 setup.py build -- ${{ matrix.instruction_set_flag }} -DCMAKE_C_COMPILER=${{ matrix.c_compiler }} -DCMAKE_CXX_COMPILER=${{ matrix.cxx_compiler }}
+      run: python setup.py build -- ${{ matrix.instruction_set_flag }} -DCMAKE_C_COMPILER=${{ matrix.c_compiler }} -DCMAKE_CXX_COMPILER=${{ matrix.cxx_compiler }}
 
     - name: Test
-      shell: bash
       # Execute tests defined by the CMake configuration.  
       # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
-      run:  ctest --test-dir _skbuild/*/cmake-build/libdistopia
\ No newline at end of file
+      run:  ctest --test-dir _skbuild/*/cmake-build/libdistopia
+
+  python-matrix:
```

Comment:
Giving up for the evening, but the reason I'm doing this is to have a job set that specifically goes through the "modern" wheel building interface that happens when you call `pip install` (at least when you have a `pyproject.toml` defined).